src/main.rs

```diff
@@ -32,78 +57,93 @@ fn main() -> Result<(), Box<dyn Error>> {
     // Render loop in a separate thread
     let (render_tx, render_rx) = mpsc::channel();
     let render_handle = thread::spawn(move || {
-        let mut last_frame = frame::new_frame();
-        let mut stdout = io::stdout();
-        render::render(&mut stdout, &last_frame, &last_frame, true);
-        loop {
-            let curr_frame = match render_rx.recv() {
-                Ok(x) => x,
-                Err(_) => break,
-            };
-            render::render(&mut stdout, &last_frame, &curr_frame, false);
-            last_frame = curr_frame;
-        }
+        render_screen(render_rx);
     });
 
     // Game loop
     let mut player = Player::new();
     let mut instant = Instant::now();
     let mut invaders = Invaders::new();
     let mut score = Score::new();
+    let mut menu = Menu::new();
+    let mut in_menu = true;
 
     'gameloop: loop {
         // Per-frame init
         let delta = instant.elapsed();
         instant = Instant::now();
         let mut curr_frame = new_frame();
 
-        // Input
-        while event::poll(Duration::default())? {
-            if let Event::Key(key_event) = event::read()? {
-                match key_event.code {
-                    KeyCode::Left => player.move_left(),
-                    KeyCode::Right => player.move_right(),
-                    KeyCode::Char(' ') | KeyCode::Enter => {
-                        if player.shoot() {
-                            audio.play("pew");
+        if in_menu {
+            // Input handlers for the menu
+            while event::poll(Duration::default())? {
+                if let Event::Key(key_event) = event::read()? {
+                    match key_event.code {
+                        KeyCode::Up => menu.change_option(true),
+                        KeyCode::Down => menu.change_option(false),
+                        KeyCode::Char(' ') | KeyCode::Enter => {
+                            if menu.selection == 0 {
+                                in_menu = false;
+                            } else {
+                                break 'gameloop;
+                            }
                         }
+                        _ => {}
                     }
-                    KeyCode::Esc | KeyCode::Char('q') => {
-                        audio.play("lose");
-                        break 'gameloop;
+                }
+            }
+            menu.draw(&mut curr_frame);
+        } else {
```

Comment:
If we finish the rendering logic and add a `continue` here, then we can get rid of the whole `else` block, decrease the indent level for all the rest of the main game loop, and not worry about whether we are in a menu later on.

```suggestion
            let _ = render_tx.send(curr_frame);
            thread::sleep(Duration::from_millis(1));
            continue;
        }
```